src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -830,6 +830,13 @@ trait Contexts { self: Analyzer =>
     private def isQualifyingImplicit(name: Name, sym: Symbol, pre: Type, imported: Boolean) =
       sym.isImplicit &&
       isAccessible(sym, pre) &&
+      !({
+        // [eed3si9n] ideally I'd like to do this: val fd = settings.future && sym.isDeprecated
+        // but implicit caching currently does not report sym.isDeprecated correctly.
```

Comment:
@hrhino In this position, I originally wrote `val fd = settings.future && sym.isDeprecated`, but sym here didn't return `isDeprecated` at least with the tests that I was using. I suspect https://github.com/scala/scala/blob/v2.13.0-M3/src/reflect/scala/reflect/internal/Symbols.scala#L3426-L3434 caching without the annotation.